### PR TITLE
feat: implement dynamic sitemap generation

### PR DESCRIPTION
**Pull request by kiemflo**
Based on #1550 

TLDR: Google couldn't find marketplace items because there was no sitemap + a sitemap is very useful anyway.

This PR introduces a sitemap generation script executable by the website as a build step, similar to the solution `solidJS` has: https://github.com/solidjs/solid-site/blob/main/scripts/generateSitemap.js

### What pages to track?

We have dynamic content -> hard to generate sitemaps for and there is no good, well maintained solution for it.

Therefore, I have introduced a simple array we can update when we introduce new pages. For all pages that are important atm, I have implemented separate methods to track all pages, specifically for dynamic pages. It is the best way to maintain the website's SEO imo.

### How to generate?
from the repo root execute: `npm run build:sitemap --w=@inlang/website`

Screenshot 2023-11-01 at 3 50 17 PM

### CODE REVIEW

**Comment by samuelstroschein** [comment]
@floiem why don;t you include the sitemap building in the build script itself? An extra build step has the risk of becoming unused because ppl use `npm run build` alone

**Comment by floiem** [comment]
> @floiem why don;t you include the sitemap building in the build script itself? An extra build step has the risk of becoming unused because ppl use `npm run build` alone

@samuelstroschein because we don't always need to update the sitemap, just when new pages are getting added or new marketplace items, that's why :D

**Comment by samuelstroschein** [comment]
> @floiem because we don't always need to update the sitemap, just when new pages are getting added or new marketplace items, that's why :D

So the script needs be manually invoked? If so, what system is in place to inform people that the script needs to be run? 

It seems simpler to just run the script on every build and thereby ensure that the sitemap is never out of date.
